static/src/javascripts/projects/common/modules/commercial/membership-engagement-banner.js

```diff
@@ -202,14 +199,11 @@ const canShow = (): Promise<boolean> => {
         return Promise.resolve(false);
     }
     return getBannerParams().then(params => {
-        const userHasSeenEnoughArticles: boolean =
-            getVisitCount() >= params.minArticlesBeforeShowingBanner;
-        const bannerIsBlockedForEditorialReasons = pageShouldHideReaderRevenue();
-
         if (
-            userHasSeenEnoughArticles &&
-            !bannerIsBlockedForEditorialReasons &&
-            userIsInCorrectCohort(params.userCohort)
+            canShowBannerSync(
+                params.minArticlesBeforeShowingBanner,
```

Comment:
as discussed, actually we shouldn't check minArticlesBeforeShowingBanner in the canRun as we would want to put people in a cohort that were shown the banner in a delayed way, otherwise if we had a test "show banner after 3 impressions" and another "show banner straight away" we wouldn't get anyone in the 3 impressions test unless they already have 2 impressions before the 0 impressions test started.
Maybe this is all a result of the fact that we're trying to use AB tests to show different banners to different people.

Reply:
of course this isn't an issue for this campaign as we're setting them all the to same impressions required (zero) but we need to consider in future.

Reply:
Good point. We need to revisit how we want to use these parameters and how to run different banners concurrently (after this campaign)